databases/functions/users/get-users-karma-score.js

```diff
@@ -0,0 +1,19 @@
+/**
+ *
+ * @param {import('sequelize').Model} userModel
+ * @param {String} userId
+ */
+const getUsersKarmaScore = async (userModel, userIds = []) => {
+  if (userIds.length === 0) throw new Error('User ids cannot be empty');
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The function `getUsersKarmaScore` throws an error when the `userIds` array is empty. This might not be the best approach as it could lead to unnecessary exceptions in your application. Instead, you could return an empty array when no user IDs are provided. This would make the function more resilient and easier to use because callers wouldn't need to handle exceptions for a normal case.

```diff
- if (userIds.length === 0) throw new Error('User ids cannot be empty');
+ if (userIds.length === 0) return [];
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->